lesson16/js/main.js

```diff
@@ -0,0 +1,169 @@
+}
+
+function categoryTabInitial(){
+	const firstTab = document.querySelectorAll('.tab-button')[0];
+	firstTab.setAttribute('aria-selected','true');
+	firstTab.setAttribute('tabindex','0');
+}
+
+function renderNewsArticle(data){
+	const newsContentArea = document.querySelector('.news-area');
+	const newsLists = document.createElement('ul');
+	newsLists.classList.add('news-list');
+	const fragment = document.createDocumentFragment();
+
+	const displayedResults = 5;
+	for (let i = 0; i < displayedResults; ++i){
+		const li = document.createElement('li');
+		li.classList.add('news-item');
+
+		const anchor = document.createElement('a');
+		anchor.classList.add('news-link');
+		anchor.href = data[i].link;
+		anchor.textContent = data[i].title;
+		fragment.appendChild(li).appendChild(anchor);
+	}
+
+	newsContentArea.appendChild(newsLists).appendChild(fragment);
+};
+
+function resetArticle(){
+	const newsLists= document.querySelector('.news-list');
+	newsLists.remove();
+}
+
+function displayErrorMassage(error){
+	const getNewsArea = document.querySelectorAll('.news-area');
+	for(let i = 0; i < getNewsArea.length; i++){
+		const createTextBox = document.createElement('p');
+		createTextBox.textContent = error;
+		getNewsArea[i].appendChild(createTextBox);
+	}
+}
+
+async function fetchNewsData() {
+	const NEWS_DATA_URL = 'https://api.json-generator.com/templates/2PqhEvPcqUZW/data?access_token=b0154huvd1stffra1six9olbgg34r4zofcqgwzfl';
+	try {
+		const response = await fetch(NEWS_DATA_URL);
+		if(response.ok){
+			const json = await response.json();
+			return json;
+		} else{
+			throw new Error('Network Error');
+		}
+	}
+	catch (error){
+		console.error('Failed to get the data. error:',error);
+		displayErrorMassage(error);
+	}
+}
+
+async function callCategry(){
+	const data = await fetchNewsData();
+	const categoryData = data.category;
+
+	if(categoryData){
+		renderCategoryTab(categoryData);
+		categoryTabInitial()
+	}
+}
+
+async function callAllArticle(){
+	showLoadingImg();
+	try{
+		const data = await fetchNewsData();
+		const articleData = data.article;
+		Object.assign(neswArticle,articleData);
+		renderNewsArticle(articleData);
+	} catch (error) {
+		displayErrorMassage(error);
+	}finally{
+		hideLoadingImg();
+	}
+}
+
+function callSerectArticle(e){
+	const selectedTab = e.target.textContent;
+	const sortData = neswArticle.filter((filterData) => {
+		return filterData.category === selectedTab;
+	});
+	renderNewsArticle(sortData);
+}
+
+function init(){
+	createTabContent()
+	callCategry();
+	callAllArticle()
+}
+
+init();
+
+function changeTabs(e) {
+	const tabs = document.querySelectorAll('[role="tab"]');
+	const selectedTab = e.target;
+	tabs.forEach((target) => target.setAttribute('aria-selected', false));
+	selectedTab.setAttribute('aria-selected', true);
+
+	const selectCategory = selectedTab.textContent;
+
+	if(selectCategory === 'ニュース'){
+		resetArticle()
+		renderNewsArticle(neswArticle);
+	}else{
```

Comment:
MUST
Please include information in the json file as to which articles will be initially displayed.

Reply:
Thank you for the review.

With the change in json.
The code has been substantially modified.

The initial display is handled in the tabContentsInitialDisplay() function.

[47c2157](https://github.com/nyattsu72/hanson-frontend-cource/pull/21/commits/47c2157556b010f1a6336ddb5e14529ee753a612)